addons/sale_purchase/models/sale_order.py

```diff
@@ -277,7 +277,8 @@ def _purchase_service_create(self, quantity=False):
         for line in self:
             line = line.with_company(line.company_id)
             # determine vendor of the order (take the first matching company and product)
-            suppliers = line.product_id._select_seller(quantity=line.product_uom_qty, uom_id=line.product_uom)
+            suppliers = line.product_id.with_company(line.company_id)._select_seller(
+                partner_id=self._context.get('supplier_map', {}).get(line.id), quantity=line.product_uom_qty, uom_id=line.product_uom)
```

Comment:
Unnecessary `with_company` here, the company of `line` environment is already the right one (see line 278)